### DISCUSSION TITLE
Creating policies and fetching data based upon a team_id instead of auth.user_id()

### DISCUSSION THREAD

**Discussion by jon-emble** [supabase/supabase#19443] [Questions]
I'm currently trying to build in a team level of functionality into my app, using Supabase's database and auth, and it's proving to be a bit of a nightmare.

In short, why can't I add an additional column onto the auth.users table, such as 'team_id', and then that gets returned into the users JWT as part of the auth process?

My current database structure is:

Table 1 - teams
- team_id: uuid
- team_name: text 

Table 2 - team_members
- user_id: auth.uid()
- team_id: foreign key of team_id from teams

Table 3 - tasks
- task_id: uuid
- team_id: foreign key of team_id from teams
- title: text
- description: text

Now when I want to fetch my list of tasks, as a signed in user, I have to do...

SELECT team_id FROM team_members WHERE user_id = auth.uid()
SELECT * from TASKS where team_id = team_id

Essentially, two select calls, and since I'm using the JavaScript library, two API calls.

In this situation, I'd like to join on the TASKS table, based upon the team_id returned alongside the user_id supplied via auth.uid(). However, I can't seem to do this (I may be seriously misunderstanding here, I'm not a SQL expert), and can only join the tables if I know what value I'm joining on ahead of time.

But since there isn't really a way of fetching this as part of middleware or getting the users session, I have to first fetch the team_id, and then the respective tasks.

On top of that, the documentation around creating policies is pretty unclear. I want to use RLS on my tasks table, based upon the user's team_id, and only fetch data if the user's team_id matches the tasks team_id, but again, I can't get access to the team_id until I have the team_id to hand - in this case, how am I meant to secure my table?

Is there a better way to do this using the Supabase JS library? I feel like I may be missing something really obvious, and can't really find anything online about how to create team structure using Supabase.

**Comment by GaryAustin1** [comment]
I've been working on this repository which shows the way to do properties like team_id or role using a table and RLS policies.  https://github.com/GaryAustin1/custom-properties   It basically would work with your teams and teams_members tables by just editing the functions to your table and column names and then using the RLS adjusted for the same names.


Users normally use https://github.com/supabase-community/supabase-custom-claims if you want the claim to be part of the jwt.

There is no way to store extra data in auth.users other than using the app metadata claim and column.
